provider/aws/registries.go

```diff
@@ -36,8 +57,12 @@ func (p *AWSProvider) RegistryAdd(server, username, password string) (*structs.R
 	// validate login
 	switch {
 	case regexpECRHost.MatchString(server):
+		if t := isOurOwnECR(p, server); t == true {
+			return nil, fmt.Errorf("can't add own ECR host as a private registry: %s", server)
```

Comment:
@MiguelMoll From what I see based on `egrep -r Errorf ./*  | grep -i rack` it looks like it's lowercase almost everywhere.